### PR TITLE
Bugfix/bgr 3615 verification reusable

### PR DESCRIPTION
**Pull request by dokkaebi**

### CODE REVIEW

File: apps/badgeuser/tests/tests.py [hunk 1]
```diff
@@ -485,6 +487,28 @@ def test_user_can_resend_verification_email(self):
...
+        self.assertEqual(response.status_code, 201)
+
+        # receive verification email
+        self.assertEqual(len(mail.outbox), 1)
+        verify_url = re.search("(?P<url>/v1/[^\s]+)", mail.outbox[0].body).group("url")
+
+        # verify the email address successfully
+        response = self.client.get(verify_url)
+        self.assertEqual(response.status_code, 302)
+        self.assertIn('authToken', response['location'])
```
Comment (nqthqn) [inline apps/badgeuser/tests/tests.py]: For parity, add `self.assertNotIn('authError', response['location'])`

File: apps/badgeuser/api.py [hunk 2]
```diff
@@ -439,6 +439,12 @@ def get(self, request, **kwargs):
                                                     "user. You have been sent a new link. Please check your email "
                                                     "and try again.")
 
+        if email_address.verified:
+            logger.event(badgrlog.EmailConfirmationAlreadyVerified(
+                request, email_address=email_address, token=token))
```
Comment (nqthqn) [inline apps/badgeuser/api.py]: I don't think we should be logging PII and a token. Could this be anonymized and serve the same benefit?

Comment (dokkaebi) [reply apps/badgeuser/api.py]: Modeled this on the other cases (e.g., `badgrlog.InvalidEmailConfirmationToken`). If this is a problem, we should reevaluate all of this logging.

Comment (nqthqn) [reply apps/badgeuser/api.py]: @ottonomy Care to chime in?

Comment (ottonomy) [reply apps/badgeuser/api.py]: Right, that's a bigger question. Accepting this for now.

**Pull request merged**